torch/csrc/jit/script/exit_transforms.cpp

```diff
@@ -419,12 +430,25 @@ struct ExitTransformer {
     // exit values. since the exit does not extend outside this block,
     // update returned exit to false. then, reset the target_block to whatever
     // it was previously
-    // this will occur with a Loop block when transforming LoopContinuations
-    // and a Graph/Closure block when transforming ReturnStmts.
     if (target_block_ == block) {
-      TORCH_INTERNAL_ASSERT(getExitStatus(exit_pair) == ExitStatus::WILL);
-      registerBlockOutputs(block, exit_pair.exitValues());
-      exit_pair = ExitPair(false_val_, std::vector<Value*>({}));
+      if (getExitStatus(exit_pair) == ExitStatus::MIGHT) {
```

Comment:
Could you please elaborate a bit on what's going on here?

Reply:
added comment